ckanext/xloader/plugin.py

```diff
@@ -94,23 +97,54 @@ def configure(self, config_):
                     )
                 )
 
-    # IResourceUrlChange
+    # IDomainObjectModification
+
+    def notify(self, entity, operation):
+        # type: (ckan.model.Package|ckan.model.Resource, DomainObjectOperation) -> None
+        """
+        Runs before_commit to database for Packages and Resources.
+        We only want to check for changed Resources for this.
+        We want to check if values have changed, namely the url.
+        See: ckan/model/modification.py.DomainObjectModificationExtension
+        """
+        if operation != DomainObjectOperation.changed \
+        or not isinstance(entity, Resource):
+            return
+
+        # If the resource requires validation, stop here if validation
+        # has not been performed or did not succeed. The Validation
+        # extension will call resource_patch and this method should
+        # be called again. However, url_changed will not be in the entity
+        # once Validation does the patch.
+        if 'validation' in toolkit.config.get('ckan.plugins', []) and \
+        toolkit.asbool(toolkit.config.get('ckanext.xloader.requires_validation')):
+            if entity.__dict__.get('extras', {}).get('validation_status', None) != 'success':
+                log.debug("Skipping xloading resource %s because "
+                          "resource did not pass validation yet.", entity.id)
+                return
+        elif not getattr(entity, 'url_changed', False):
+            return
 
-    def notify(self, resource):
         context = {
             "ignore_auth": True,
         }
         resource_dict = toolkit.get_action("resource_show")(
             context,
             {
-                "id": resource.id,
+                "id": entity.id,
             },
         )
         self._submit_to_xloader(resource_dict)
 
     # IResourceController
 
     def after_resource_create(self, context, resource_dict):
+        if 'validation' in toolkit.config.get('ckan.plugins', []) and \
```

Comment:
This will not catch variations on ckanext-validation that use different plugin definitions. Is it feasible to instead check for the presence of one of the ckanext-validation actions, like `resource_validation_show`?

Reply:
@ThrawnCA ah makes sense, in the case that a developer subclasses the whole Validation Plugin Class? I have added the new check for the `resource_validation_show` action